### ISSUE TITLE
Aria labels for table header sorting arrow buttons

### ISSUE THREAD

**Issue by edenhernandez-suse** [rancher/dashboard#8191]
## Context
Our tables have the option to sort by every single column header category:

![image](https://user-images.githubusercontent.com/120640736/219369432-60d8d5be-b21e-4d0f-bacc-f5c5ffa81d86.png)

## The bug
The sorting option is not 100% accessible. It is possible to activate it, but that option is not announced by screen readers. There is also no information about which column the table is currently sorted by.

## Expected Result
Ideally, the table headers should be/have a button to announce that they're clickable (with a caption, for instance). Aria labels can be used to announce how the table is currently sorted.

An example from [W3C sortable table](https://w3c.github.io/aria-practices/examples/table/sortable-table.html) - Here the [example in a CodePen](https://codepen.io/edenhernandez-suse/pen/rNZaPzK)

![image](https://user-images.githubusercontent.com/120640736/219371423-f91bce90-45d9-44fe-a713-9387d39a006c.png)

Part of #7968

**Comment by izaac** [comment]
Rancher version: `v2.12-9d8a81e5882a1c435fd4b5f68428e59dd69d73d1-head`
Rancher UI version (dashboard): `master c0d31f3`

Using Narrator the sortable voice is able to describe the sorting order using MS Edge and Chrome. On Firefox however for some reason it just describe it as "<Name> header, row" but no sorting order.

I was able to access those texts more consistently using NVDA, it worked on both browsers. So the Narrator may have some integration limits on Firefox. Moving to done ✅
